gapis/gfxapi/gles/gles.api

```diff
@@ -515,34 +515,47 @@ sub void ApplyStaticContextState(ref!Context ctx, ref!StaticContextState staticS
 
 sub void ApplyDynamicContextState(ref!Context ctx, ref!DynamicContextState dynamicState) {
   if (dynamicState != null) {
-    backbuffer := ctx.Objects.Framebuffers[0]
-    color_id := backbuffer.ColorAttachments[0].Renderbuffer.ID
-    depth_id := backbuffer.DepthAttachment.Renderbuffer.ID
-    stencil_id := backbuffer.StencilAttachment.Renderbuffer.ID
-
-    color_buffer := ctx.SharedObjects.Renderbuffers[color_id]
-    color_buffer.Width = dynamicState.BackbufferWidth
-    color_buffer.Height = dynamicState.BackbufferHeight
-    color_buffer.InternalFormat = dynamicState.BackbufferColorFmt
-    ctx.SharedObjects.Renderbuffers[color_id] = color_buffer
-
-    depth_buffer := ctx.SharedObjects.Renderbuffers[depth_id]
-    depth_buffer.Width = dynamicState.BackbufferWidth
-    depth_buffer.Height = dynamicState.BackbufferHeight
-    depth_buffer.InternalFormat = dynamicState.BackbufferDepthFmt
-    ctx.SharedObjects.Renderbuffers[depth_id] = depth_buffer
-
-    stencil_buffer := ctx.SharedObjects.Renderbuffers[stencil_id]
-    stencil_buffer.Width = dynamicState.BackbufferWidth
-    stencil_buffer.Height = dynamicState.BackbufferHeight
-    stencil_buffer.InternalFormat = dynamicState.BackbufferStencilFmt
-    ctx.SharedObjects.Renderbuffers[stencil_id] = stencil_buffer
-
-    if dynamicState.ResetViewportScissor {
-      ctx.FragmentOperations.Scissor.Box.Width = dynamicState.BackbufferWidth
-      ctx.FragmentOperations.Scissor.Box.Height = dynamicState.BackbufferHeight
-      ctx.Rasterization.Viewport.Width = dynamicState.BackbufferWidth
-      ctx.Rasterization.Viewport.Height = dynamicState.BackbufferHeight
+    if 0 in ctx.Objects.Framebuffers {
```

Comment:
I agree that any sequence of commands should not cause panic, but you have to convince me that is the case here.  With eglMakeCurrent missing this would just result in no context being bound, which would make other commands gracefully stop.

I agree that with quick scan thought the commands, you might want to cancel the unneeded ones.  Let's say you have 1000 requests pending and currently processing the first one, but only needing the last one.  I say you should finish the first one, skip the 998 which have not even started yet, and then do the last one.  If you really want to try hard, I say you can check for cancellation in the top-most loop so you bail out once then next atom is fully processed.  I fail to see why it is important to cancel immediately in the most nested function and then deal with the error propagation and state inconsistency problems.  It sounds to me like yet another really complicated solution for really simple problem.